### PR TITLE
rbac: support reading service definition from a file for the upgrade command

### LINKED ISSUE #12075: istioctl experimental rbac can incorrectly reports 'deny'
To reproduce, I first deployed httpbin in the usual way.  Follow https://istio.io/docs/tasks/traffic-management/ingress/ or just do

```
kubectl apply -f samples/httpbin/httpbin.yaml
kubectl apply -f samples/httpbin/httpbin-gateway.yaml
```

I verified it worked: `curl -s -o /dev/null -w "%{http_code}" http://$INGRESS_HOST:31380`

I did `istioctl experimental rbac can GET httpbin /` and the response was ```
deny
```

I tried various options such as `--user default` and `--user cluster.local/ns/default/sa/default`.  I always received "deny".

The documentation shows `... GET rating /v1/health` and probably `ratings` is intended instead of `rating`.

### PR DESCRIPTION
**Pull request by yangminzhu**
This PR does the following:
- Add an option `--service` to allow reading service definition from a yaml file, this also makes it possible to test the `upgrade` command in unit test.
- Remove the deprecated and broken `rbac can` command (Fixes #12075), the same function will be provided in the `auth` group
- Move the `upgrade` command to under the `auth` group

### CODE REVIEW

**Comment by yangminzhu** [comment]
@esnible @pitlv2109 The PR is ready for review, could you take a look? Thanks

**Comment by pitlv2109** [comment]
Thank you for working on this. I will look at it later today.

File: istioctl/cmd/auth.go [hunk 1]
```diff
@@ -158,4 +223,8 @@ func init() {
 		"Show additional information (e.g. SNI and ALPN)")
 	checkCmd.PersistentFlags().StringVarP(&configDumpFile, "file", "f", "",
 		"Check the TLS/JWT/RBAC setting from the config dump file")
+	upgradeCmd.PersistentFlags().StringVarP(&v1PolicyFile, "file", "f", "",
+		"Authorization policy file")
+	upgradeCmd.PersistentFlags().StringSliceVarP(&serviceSpecs, "service", "s", []string{},
+		"Service specification that provides the mapping relationship between service name and workload labels")
```
Comment (pitlv2109) [inline istioctl/cmd/auth.go]: Maybe mention the input is a yaml file with service spec. When I first read it, I thought it was something like `service-name: workload-label` by the "provides the mapping relationship between service name and workload labels"

File: istioctl/pkg/auth/upgrade.go [hunk 2]
```diff
@@ -41,12 +46,13 @@ type WorkloadLabels map[string]string
...
 	IstioConfigStore model.ConfigStore
 	K8sClient        *kubernetes.Clientset
+	ServiceFiles     []string
 	// RoleNameToWorkloadLabels maps the ServiceRole name to the map of service name to the workload labels
 	// that the service pointing to.
 	// We could have remove the `service` key layer to make it ServiceRole name maps to workload labels.
 	// However, this is needed for unit tests.
 	RoleNameToWorkloadLabels map[string]ServiceToWorkloadLabels
-	RbacFile                 string
+	V1PolicyFile             string
```
Comment (pitlv2109) [inline istioctl/pkg/auth/upgrade.go]: Why changing the name? Are we planning on using this `upgrader` for authN as well?

File: istioctl/cmd/auth_test.go [hunk 3]
```diff
@@ -23,30 +23,56 @@ import (
...
-		if err != nil {
-			t.Errorf("%s: unexpected error: %s", c.name, err)
-		} else {
-			util.CompareContent(out.Bytes(), c.goldenFilename, t)
-		}
+		runCommandAndCheckGolden(c.name, command, c.golden, t)
+	}
+}
+
+func TestAuthUpgrade(t *testing.T) {
```
Comment (pitlv2109) [inline istioctl/cmd/auth_test.go]: Maybe change the name to TestAuthzUpgrade or TestRbacUpgrade? Later if we want to test AuthN upgrade, we can just create another function to decouple the tests?

File: istioctl/cmd/auth.go [hunk 4]
```diff
@@ -19,18 +19,21 @@ import (
...
 	"istio.io/istio/istioctl/pkg/util/configdump"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/log"
 )
 
 var (
 	printAll       bool
 	configDumpFile string
+	v1PolicyFile   string
+	serviceSpecs   []string
```
Comment (pitlv2109) [inline istioctl/cmd/auth.go]: Sorry if I missed it, but does each serviceSpecs file contain the whole service config (name, namespace, etc.) or just the spec field? Looks like the second case based on the name?

File: istioctl/cmd/auth_test.go (hunk 3 above)
Comment (yangminzhu) [reply istioctl/cmd/auth_test.go]: I would like to keep this name, in the future, we can have both authN and authZ test cases here, they can even appear in the same input file. From the end user, they only need to invoke the `upgrade` command and it works for both authN and authZ.

File: istioctl/cmd/auth.go (hunk 4 above)
Comment (yangminzhu) [reply istioctl/cmd/auth.go]: Changed to `serviceFiles`

File: istioctl/pkg/auth/upgrade.go (hunk 2 above)
Comment (yangminzhu) [reply istioctl/pkg/auth/upgrade.go]: Yes, I'm planning to make the `upgrade` support both AuthN and AuthZ.

/cc @diemtvu

File: istioctl/cmd/auth.go (hunk 1 above)
Comment (yangminzhu) [reply istioctl/cmd/auth.go]: Updated the comment to make it clear it's k8s service resource.

**Comment by pitlv2109** [comment]
/lgtm

I think in the future if we want to also use `upgrade` for authN, we should separate the logic code in `upgrade.go` for authN and authZ into two files.

**Comment by yangminzhu** [comment]
@esnible Could you take a look at this? Thank you.

**Pull request merged**